src/pages/workspaces/workspace/analysis/modals/AnalysisModal.ts

```diff
@@ -209,30 +239,48 @@ export const AnalysisModal = withDisplayName('AnalysisModal')(
     }, [
       renderToolButtons(),
       h(Dropzone, {
-        accept: `.${tools.Jupyter.ext.join(', .')}, .${tools.RStudio.ext.join(', .')}`,
+        accept: `.${runtimeTools.Jupyter.ext.join(', .')}, .${runtimeTools.RStudio.ext.join(', .')}`,
         style: { flexGrow: 1, backgroundColor: colors.light(), height: '100%' },
         activeStyle: { backgroundColor: colors.accent(0.2), cursor: 'copy' },
         onDropRejected: () => reportError('Not a valid analysis file',
-          `The selected file is not one of the supported types: .${tools.Jupyter.ext.join(', .')}, .${tools.RStudio.ext.join(', .')}. Ensure your file has the proper extension.`),
+          `The selected file is not one of the supported types: .${runtimeTools.Jupyter.ext.join(', .')}, .${runtimeTools.RStudio.ext.join(', .')}. Ensure your file has the proper extension.`),
         onDropAccepted: files => {
-          const tool = !!googleProject ? tools[getToolFromFileExtension(files.pop().path)] : runtimeTools.JupyterLab
+          let tool
```

Comment:
We really try to stay away from mutation in the codebase...

Could this instead read 
```
const tool = isGoogleWorkspaceInfo(workspace.workspace) ? _.get(getToolFromFileExtension(files.pop().path), tools) : runtimeTools.JupyterLab
```

If so, it might be worth renaming `getToolFromFileExtension` to `getToolLabelFromFileExtension`

Reply:
I renamed a few of the incorrectly named functions from tool-utils.